lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2438,6 +2490,63 @@ bool CppEmitter::hasBlockLabel(Block &block) {
   return blockMapper.count(&block);
 }
 
+template <typename ElTy>
+static bool hasSameDenseValue(DenseIntElementsAttr dense,
+                              std::string &firstValue) {
+  SmallVector<ElTy> denseValues;
+  for (APInt elem : dense) {
+    denseValues.push_back(elem.getLimitedValue());
+  }
+  ElTy firstV = denseValues[0];
+  if (llvm::all_of(denseValues, [firstV](ElTy v) { return v == firstV; })) {
+    firstValue = std::to_string(firstV);
+    return true;
+  }
+  return false;
+}
+
+static bool hasSameDenseValueOfFloat(DenseFPElementsAttr dense,
```

Comment:
For getting splat value, I think you can leverage isSplat() and getSplatValue(). See https://mlir.llvm.org/doxygen/classmlir_1_1DenseElementsAttr.html.

Reply:
Thanks, James.